### ISSUE TITLE
Age Restricted Youtube Video Support

### ISSUE THREAD

**Issue by bencorryp** [codetheweb/muse#997]
Is it possible to allow the bot to search (and play) age-restricted content on youtube? Could not find a setting on the generated API key to allow for age restricted content to be viewed.

**Comment by codetheweb** [comment]
No, but happy to accept a PR for this.

**Comment by jogerj** [comment]
Afaik there's no way of directly accessing restricted videos from the YouTube API without being logged in. In other words, you'll have to pass a YouTube account credentials by exporting your browser cookies and passing it to the client

**Comment by Happyllama25** [comment]
What about the ytdl library? I've used ytdlp (Python fork) in some personal projects and age-restricted video downloads work as intended. 
Granted, I'm not sure how they do it, but it's probably not account credentials.

**Comment by jogerj** [comment]
@Happyllama25 I've worked with yt-dlp before. [Passing cookies to the client is the way](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp). It's straight forward in [code](https://github.com/Just-Some-Bots/MusicBot/commit/894541303bff1a6aeacbdf1a5ac7da88ac3b02f2) too, but since this project is written in typescript and doesn't rely on yt-dlp at all, we'll have to do on our own. But the proof of concept above should work.

**Comment by Happyllama25** [comment]
> It does work, sort-of

Yes, it works for me on my bot, something I did notice though is it seems to take longer to download an age-restricted video than a non-restricted one - not 100% sure, I have not done a 1:1 test of this.

![image](https://github.com/codetheweb/muse/assets/49815691/8cd1a855-46d4-4ceb-a5fc-23949af45521)
